packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -134,536 +87,971 @@ void main() {
-    jsonObject = json.decode(service.getSelectedWidget(null, 'my-group'));
-    expect(jsonObject, isNot(contains('createdByLocalProject')));
-    creationLocation = jsonObject['creationLocation'];
-    expect(creationLocation, isNotNull);
-    // This RichText widget is created by the build method of the Text widget
-    // thus the creation location is in text.dart not basic.dart
-    final List<String> pathSegmentsFramework = Uri.parse(creationLocation['file']).pathSegments;
-    expect(pathSegmentsFramework.join('/'), endsWith('/packages/flutter/lib/src/widgets/text.dart'));
-
-    // Strip off /src/widgets/text.dart.
-    final String pubRootFramework = '/' + pathSegmentsFramework.take(pathSegmentsFramework.length - 3).join('/');
-    service.setPubRootDirectories(<Object>[pubRootFramework]);
-    expect(json.decode(service.getSelectedWidget(null, 'my-group')), contains('createdByLocalProject'));
-    service.setSelection(elementA, 'my-group');
-    expect(json.decode(service.getSelectedWidget(null, 'my-group')), isNot(contains('createdByLocalProject')));
-
-    service.setPubRootDirectories(<Object>[pubRootFramework, pubRootTest]);
-    service.setSelection(elementA, 'my-group');
-    expect(json.decode(service.getSelectedWidget(null, 'my-group')), contains('createdByLocalProject'));
-    service.setSelection(richText, 'my-group');
-    expect(json.decode(service.getSelectedWidget(null, 'my-group')), contains('createdByLocalProject'));
-  }, skip: !WidgetInspectorService.instance.isWidgetCreationTracked()); // Test requires --track-widget-creation flag.
+      );
+      final Element elementA = find.text('a').evaluate().first;
+
+      await service.testExtension('setPubRootDirectories', <String, String>{});
+      service.setSelection(elementA, 'my-group');
+      Map<String, Object> jsonObject = await service.testExtension('getSelectedWidget', <String, String>{'arg': null, 'objectGroup': 'my-group'});
+      Map<String, Object> creationLocation = jsonObject['creationLocation'];
+      expect(creationLocation, isNotNull);
+      final String fileA = creationLocation['file'];
+      expect(fileA, endsWith('widget_inspector_test.dart'));
+      expect(jsonObject, isNot(contains('createdByLocalProject')));
+      final List<String> segments = Uri.parse(fileA).pathSegments;
+      // Strip a couple subdirectories away to generate a plausible pub root
+      // directory.
+      final String pubRootTest = '/' + segments.take(segments.length - 2).join('/');
+      await service.testExtension('setPubRootDirectories', <String, String>{'arg0': pubRootTest});
+
+      service.setSelection(elementA, 'my-group');
+      expect(await service.testExtension('getSelectedWidget', <String, String>{'objectGroup': 'my-group'}), contains('createdByLocalProject'));
+
+      await service.testExtension('setPubRootDirectories', <String, String>{'arg0': '/invalid/$pubRootTest'});
+      expect(await service.testExtension('getSelectedWidget', <String, St
```

Comment:
The only delta from the previous commit other than indenting by two spaces as this is now a static method is the addition of this ext.flutter.inspector.show test.